presto-main/src/test/java/com/facebook/presto/operator/scalar/TestDateTimeFunctions.java

```diff
@@ -763,6 +796,21 @@ public void testDateParse()
         assertFunction("date_parse('2013-05', '%Y-%m')",
                 TimestampType.TIMESTAMP,
                 toTimestamp(new DateTime(2013, 5, 1, 0, 0, 0, 0, DATE_TIME_ZONE)));
+        assertFunction("date_parse('2013-05-17th', '%Y-%m-%D')",
```

Comment:
Add a another parsing test case where it's in the middle: `%M %D, %Y` -> `November 5th, 1955`